xblock/fields.py

```diff
@@ -824,9 +833,23 @@ class String(JSONField):
     """
     MUTABLE = False
 
+    def _sanitize(self, value):
+        """
+        Remove the control characters that are not allowed in XML:
+        https://www.w3.org/TR/xml/#charsets
+        Leave all other characters.
+        """
+        if isinstance(value, unicode):
+            new_value = u''.join(ch for ch in value if unicodedata.category(ch)[0] != u'C' or ch in (u'\n', u'\r', u'\t'))
+        elif isinstance(value, str):
+            new_value = ''.join(ch for ch in value if ord(ch) >= 32 or ch in ('\n', '\r', '\t'))
+        else:
+            return value
+        return value if value == new_value else new_value
```

Comment:
Actually - the logic needs to stay as it was. When a new string is formed, it may be equivalent to the original string if there are no control characters present. In that case, the original string needs to be returned because some tests check for *object* equality instead of *string* equality. For example:
https://github.com/edx/XBlock/blob/master/xblock/test/test_fields_api.py#L128
@cpennington Is this semantic change going to cause issues anywhere?